app/src/main/java/io/gnosis/safe/ui/StartActivity.kt

```diff
@@ -1,13 +1,87 @@
 package io.gnosis.safe.ui
 
 import android.os.Bundle
+import android.view.View
+import androidx.navigation.Navigation
+import androidx.navigation.ui.setupWithNavController
+import com.google.android.material.bottomnavigation.BottomNavigationView
+import io.gnosis.data.models.Safe
 import io.gnosis.safe.R
+import io.gnosis.safe.databinding.ToolbarSafeOverviewBinding
 import io.gnosis.safe.ui.base.BaseActivity
+import io.gnosis.safe.ui.safe.SafeOverviewNavigationHandler
+import io.gnosis.safe.ui.safe.selection.SafeSelectionDialog
+import io.gnosis.safe.utils.asMiddleEllipsized
+import pm.gnosis.svalinn.common.utils.visible
+import pm.gnosis.utils.asEthereumAddressString
 
-class StartActivity : BaseActivity() {
+class StartActivity : BaseActivity(), SafeOverviewNavigationHandler {
+
+    private val toolbarBinding by lazy {
+        ToolbarSafeOverviewBinding.bind(findViewById(R.id.toolbar_container))
+    }
+    private val toolbar by lazy { findViewById<View>(R.id.toolbar) }
+    private val navBar by lazy { findViewById<BottomNavigationView>(R.id.nav_bar) }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_start)
+        toolbarBinding.safeSelection.setOnClickListener {
+            SafeSelectionDialog.show(this)
+        }
+        setupNav()
+    }
+
+    private fun setupNav() {
+        val navController = Navigation.findNavController(this, R.id.nav_host)
+        navBar.setupWithNavController(navController)
+        navController.addOnDestinationChangedListener { _, destination, _ ->
+            if (isFullscreen(destination.id)) {
+                toolbar.visibility = View.GONE
+                navBar.visibility = View.GONE
```

Comment:
Did you decide to not use motion layout? I suppose that here you would have `transitionToEnd` or `start` depending on how you defined it.

Reply:
yes, basically it is same logic, but without animated transitions. We could add motion scene at some later point.